contracts/test/OpportunityAdapterIntegration.t.sol

```diff
@@ -293,6 +306,173 @@ contract OpportunityAdapterIntegrationTest is
         );
     }
 
+    function testExecutionWithBidAndCallValueViaSwaps(
```

Comment:
I know that the scenario in this test case is complicated and I think it's worth a few lines of comment to explain what's going on. Maybe even some ascii art to show the flow